packages/hardhat-chai-matchers/test/changeEtherBalance.ts

```diff
@@ -0,0 +1,550 @@
+        });
+
+        it("Should throw when fee was not calculated correctly", async () => {
+          await expect(
+            expect(() =>
+              sender.sendTransaction({
+                to: receiver.address,
+                gasPrice: 1,
+                value: 200,
+              })
+            ).to.changeEtherBalance(sender, -200, { includeFee: true })
+          ).to.be.eventually.rejectedWith(
+            AssertionError,
+            `Expected "${
+              sender.address
+            }" to change balance by -200 wei, but it has changed by -${
+              txGasFees + 200
+            } wei`
+          );
+        });
+
+        it("Should throw when expected balance change value was different from an actual", async () => {
+          await expect(
+            expect(() =>
+              sender.sendTransaction({
+                to: receiver.address,
+                value: 200,
+              })
+            ).to.changeEtherBalance(sender, "-500")
+          ).to.be.eventually.rejectedWith(
+            AssertionError,
+            `Expected "${sender.address}" to change balance by -500 wei, but it has changed by -200 wei`
+          );
+        });
+
+        it("Should throw in negative case when expected balance change value was equal to an actual", async () => {
+          await expect(
+            expect(() =>
+              sender.sendTransaction({
+                to: receiver.address,
+                value: 200,
+              })
+            ).to.not.changeEtherBalance(sender, "-200")
+          ).to.be.eventually.rejectedWith(
+            AssertionError,
+            `Expected "${sender.address}" to not change balance by -200 wei`
+          );
+        });
+
+        it("Should pass when given zero value tx", async () => {
+          await expect(() =>
+            sender.sendTransaction({ to: receiver.address, value: 0 })
+          ).to.changeEtherBalance(sender, 0);
+        });
+      });
+
+      describe("Change balance, one contract", () => {
+        it("Should pass when expected balance change is passed as int and is equal to an actual", async () => {
+          await expect(async () =>
+            sender.sendTransaction({
+              to: contract.address,
+              value: 200,
+            })
+          ).to.changeEtherBalance(contract, 200);
+        });
+
+        it("should pass when calling function that returns half the sent ether", async () => {
+          await expect(async () =>
+            contract.returnHalf({ value: 200 })
+          ).to.changeEtherBalance(sender, -100);
+        });
+      });
+    });
+
+    describe("Transaction Callback (1559 tx)", () => {
+      describe("Change balance, one account", () => {
+        it("Should pass when expected balance change is passed as string and is equal to an actual", async () => {
+          await expect(() =>
+            sender.sendTransaction({
+              to: receiver.address,
+              maxFeePerGas: 1,
```

Comment:
Using the same `maxFeePerGas` and `maxPriorityFeePerGas` is conceptually equivalent to using `gasPrice`. I would update all the `maxFeePerGas` to another value so that these tests are a bit more robust.